types/src/system_contract_errors/auction.rs

```diff
@@ -100,11 +100,35 @@ pub enum Error {
     /// Failed to reduce total supply.
     #[fail(display = "Failed to reduce total supply")]
     MintReduceTotalSupply = 26,
+    /// Triggered when contract was unable to transfer desired amount of tokens into a delegators
+    /// purse.
+    #[fail(display = "Transfer to delegators purse error")]
+    TransferToDelegatorPurse = 27,
+    /// Triggered when contract was unable to perform a transfer to distribute validators reward.
+    #[fail(display = "Reward transfer to validator error")]
+    ValidatorRewardTransfer = 28,
+    /// Triggered when contract was unable to perform a transfer to distribute delegators rewards.
+    #[fail(display = "Rewards transfer to delegator error")]
+    DelegatorRewardTransfer = 29,
+    /// Failed to transfer desired amount while withdrawing delegators reward.
+    #[fail(display = "Withdraw delegator reward error")]
+    WithdrawDelegatorReward = 30,
+    /// Failed to transfer desired amount while withdrawing validators reward.
+    #[fail(display = "Withdraw validator reward error")]
+    WithdrawValidatorReward = 31,
+    /// Failed to transfer desired amount into unbonding purse.
+    #[fail(display = "Transfer to unbonding purse error")]
+    TransferToUnbondingPurse = 32,
+
+    #[cfg(test)]
+    #[doc(hidden)]
+    #[fail(display = "Sentinel error")]
+    Sentinel,
```

Comment:
Why do we need this variant?

Reply:
Before this change

```
#[cfg(test)]
const MAX_ERROR_VALUE
```

was hardcoded and `enum Error` values grew, but tests didnt reflect that. With this `MAX_ERROR_VALUE` would be always highest value and thus all values will be tested (if we'd extend error enum in future)